### ISSUE TITLE
renaissance-scala-kmeans_0 fails on xSolaris jdk8 hs

### ISSUE THREAD

**Issue by Haroon-Khel** [adoptium/aqa-tests#3244]
renaissance-scala-kmeans_0 from the sanity.perf suite fails on xSolaris hotspot

Failure link https://ci.adoptopenjdk.net/job/Test_openjdk8_hs_sanity.perf_x86-64_solaris/46/tapResults/

Error:
```
=============================================== Running test renaissance-scala-kmeans_0 ... 
=============================================== renaissance-scala-kmeans_0 Start Time: Wed Jan 19 17:38:29 
2022 Epoch Time (ms): 1642613909287 variation: NoOptions JVM_OPTIONS:    
renaissance-scala-kmeans_0_FAILED
```

**Comment by smlambert** [comment]
We should figure out why it appears that the benchmark does not appear to run at all.  This is very likely an issue with one or more of the scripts involved not running well on Solaris. Running with some extra verbosity turned on may help.  I would suspect run_with_affinity.sh, but the other benchmarks in the perf group also call this script, so it is odd that this particular benchmark is different.  May want to also compare the command for this test target with the command for another similar renaissance benchmark that does run and pass on solaris to see if there is just some strange typo we are dealing with...

While we investigate, it will be perfectly acceptable to exclude on this platform for now.

from [Grinder/3086](https://ci.adoptopenjdk.net/job/Grinder/3086/console)
```
15:45:12  { itercnt=1; \
15:45:12  mkdir -p "/export/home/jenkins/workspace/Grinder/aqa-tests/TKG/../TKG/output_16427115114181/renaissance-scala-kmeans_0"; \
15:45:12  cd "/export/home/jenkins/workspace/Grinder/aqa-tests/TKG/../TKG/output_16427115114181/renaissance-scala-kmeans_0"; \
15:45:12  /export/home/jenkins/workspace/Grinder/aqa-tests/TKG/../perf/run_with_affinity.sh --exec_cmd ""/export/home/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image/bin/java" -jar "/export/home/jenkins/workspace/Grinder/aqa-tests/TKG/../../jvmtest/perf/renaissance/renaissance.jar" --json ""/export/home/jenkins/workspace/Grinder/aqa-tests/TKG/../TKG/output_16427115114181/renaissance-scala-kmeans_0"/scala-kmeans.json" scala-kmeans" --test_root /export/home/jenkins/workspace/Grinder/aqa-tests/TKG/..; \
15:45:12  	if [ $? -eq 0 ] ; then echo ""; echo "renaissance-scala-kmeans_0""_PASSED"; echo ""; cd /export/home/jenkins/workspace/Grinder/aqa-tests/TKG/..; rm -f -r "/export/home/jenkins/workspace/Grinder/aqa-tests/TKG/../TKG/output_16427115114181/renaissance-scala-kmeans_0"; else echo ""; echo "renaissance-scala-kmeans_0""_FAILED"; echo ""; fi; } 2>&1 | tee -a "/export/home/jenkins/workspace/Grinder/aqa-tests/TKG/../TKG/output_16427115114181/TestTargetResult";
15:45:12  
```

**Comment by Haroon-Khel** [comment]
Also failing on sparc https://ci.adoptopenjdk.net/job/Test_openjdk8_hs_sanity.perf_sparcv9_solaris/89/

**Comment by smlambert** [comment]
@JeromeJu - we can disable this test target https://github.com/adoptium/aqa-tests/blob/master/perf/renaissance/playlist.xml#L262-L272 on the 2 solaris platforms

by adding a `<disables>` block (see this documentation for details: https://github.com/adoptium/aqa-tests/blob/master/doc/userGuide.md#exclude-a-test-against-specific-platform

```
<disables>
    <disable>
      <comment>https://github.com/adoptium/aqa-tests/issues/123456</comment>
      <platform>x86-64_solaris|sparcv9_solaris</plat>
    </disable>
  </disables>
```

**Comment by llxia** [comment]
I believe this issue was closed accidentally. We only excluded the test, but the original issue is not resolved.

**Comment by gervaisj** [comment]
## Debug

@smlambert 
@JeromeJu here is some info I discovered regarding the issue. Let me know if you prefer transferring the issue to me or you want to keep it, I don't mind either way.

> We should figure out why it appears that the benchmark does not appear to run at all. This is very likely an issue with one or more of the scripts involved not running well on Solaris. Running with some extra verbosity turned on may help. I would suspect run_with_affinity.sh

`run_with_affinity.sh` itself seems fine, but it calls (silently unfortunately) another script called `affinity.sh` which does not support Solaris/SunOS as a platform. I hacked together support for solaris in `affinity.sh` in commit 7f4aa1bbec432f8c0969c42bdba0e0773cf409a4 with dummy values. The test then ran correctly (see [Grinder 3837](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3837))

> but the other benchmarks in the perf group also call this script

Searching all Adoptium repos with GitHub, I can't find another test that uses `run_with_affinity.sh`, so I can't cross check with any other test.

However, there are 2 other tests that call `affinity.sh`, namely `dt7_sufp.sh` and `dt7_throughput.sh`. We expect those would also fail on Solaris. But running their associated test (respectively `liberty-dt7-startup` and `liberty-dt7-throughput`) in Grinder on Solaris seems to reveal another problem with the way `dt7_sufp.sh` and `dt7_throughput.sh` are called. Precisely we are using `export NAME=VALUE;` and this is not valid syntax in the solaris shell (see [here](https://www.unix.com/solaris/108368-sh-not-identifier.html)). For instance, this is the command for launching `liberty-dt7-startup`:
```shell
export SERVER_PHYSCPU_NUM=2; export SMT=true; bash $(Q)$(TEST_RESROOT)$(D)configs$(D)dt7_sufp.sh$(Q) $(TEST_RESROOT); ${TEST_STATUS}
```
And this is the output in Grinder (emphasis on error message):
```
21:12:13  export SERVER_PHYSCPU_NUM=2; export SMT=true; bash "/export/home/jenkins/workspace/Grinder/aqa-tests/TKG/../../jvmtest/perf/liberty/configs/dt7_sufp.sh" /export/home/jenkins/workspace/Grinder/aqa-tests/TKG/../../jvmtest/perf/liberty; \
21:12:13  	if [ $? -eq 0 ] ; then echo ""; echo "liberty-dt7-startup_0""_PASSED"; echo ""; cd /export/home/jenkins/workspace/Grinder/aqa-tests/TKG/..;  else echo ""; echo "liberty-dt7-startup_0""_FAILED"; echo ""; fi; } 2>&1 | tee -a "/export/home/jenkins/workspace/Grinder/aqa-tests/TKG/../TKG/output_16472203329552/TestTargetResult";
21:12:13  /bin/sh: SERVER_PHYSCPU_NUM=2: is not an identifier
```
To test the theory that it is really the syntax, I spotted another test (`jdk_swing`) that also sets an env variable on the command line, re-enabled it and launched it in Grinder on solaris
Output
```
10:12:52  export DISPLAY=:1; "/export/home/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image/bin/java" -Xmx512m -jar "/export/home/jenkins/workspace/Grinder/aqa-tests/TKG/../../jvmtest/openjdk/jtreg/lib/jtreg.jar" 
[...]
10:12:52  /bin/sh: DISPLAY=:1: is not an identifier
```

> May want to also compare the command for this test target with the command for another similar renaissance benchmark that does run and pass on solaris to see if there is just some strange typo we are dealing with...

Other renaissance benchmarks call a .jar directly. `renaissance-scala-kmeans` is the only one using `run_with_affinity.sh`

## TLDR
2 issues:
1. `affinity.sh` does not support Solaris/SunOS and fails silently in `run_with_affinity.sh`
2. `export NAME=VALUE;` is invalid syntax in the Solaris/SunOS shell
